packages/amplify-util-mock/src/CFNParser/intrinsic-functions.ts

```diff
@@ -0,0 +1,277 @@
+import { CloudFormationParseContext } from './types';
+import { isPlainObject, isInteger } from 'lodash';
+import { isString, isArray, isObject } from 'util';
+
+export function cfnJoin(
+  valNode: [string, string[]],
+  { params, conditions, resources, exports }: CloudFormationParseContext,
+  processValue
+) {
+  if (!(Array.isArray(valNode) && valNode.length === 2)) {
```

Comment:
validate ```if (valNode)``` as well and also check if 2nd item is an array as well, otherwise ```map``` later could fail.